test/unit/test_amqpclient.js

```diff
@@ -106,7 +106,7 @@ describe('AMQPClient', function() {
 
       return client.connect(mock_uri)
         .then(function () {
-          return client.createSender(queue, { name: 'queue_TX' });
+          return client.createSender(queue, { options: { name: 'queue_TX' } });
```

Comment:
I personally think just `attach` would be fine, since we're already contained inside a policy object so "attach policy" is implied. Note that this also means it could be `session.begin`, and it _also_ points out that I did a terrible separation on the `connection.options`